### PR TITLE
Heatmap

### PR DESCRIPTION
**Pull request by henrhaat**
This is still missing the instructions for the instruction page and I will do them. I couldn't figure out a good way to make the heatmap have the actual intervals as hover data.

### CODE REVIEW

File: views/heatmap.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    eq_data = earthquake_data.get_earthquake_data(session_id)
+
+    if eq_data is None or eq_data.data.shape == (0, 0):
+        return 'No uploaded data found'
+
+    start_date, end_date = eq_data.get_daterange()
+    filtered_data = eq_data.filter_by_dates(start_date, end_date)
+
+    default_x = filtered_data.get_magnitudes()
+    default_y = filtered_data.get_latitudes()
```
Comment (llmlks) [inline views/heatmap.py]: Would depth be a better default y-axis, as it's also used in the scatter plot?

**Review by llmlks — COMMENTED** [review]
Looks very good! No changes needed for the code, only a couple of improvement suggestions for the feature:
* As mentioned in the comment, would depth vs magnitude be a better default setting, as they are also used elsewhere in the app?
* The names of the axes should be visible (next to/below the relevant axis)
* Can the label of the bins be changed? If so, it would be nice to have the names of the features instead of x, y, and z. For example ('Magnitude: 1.5 \n Depth: 17 \n Count: 5'), in addition to displaying the ranges, if possible

File: views/heatmap.py (hunk 1 above)
Comment (henrhaat) [reply views/heatmap.py]: Yes, my intention is to have the default y-axis as depth, but I had a bit of trouble with one of the depth columns in one of the datasets earlier, but I will change that back. The feature names should be possible to implement, but the ranges might be trickier.

**Comment by henrhaat** [comment]
I figured out a way to do all the requested changes. The borders of the bins probably need to be rounded? Otherwise the hoverdata takes too much space

**Review by llmlks — CHANGES_REQUESTED** [review]
I think it looks great, and not taking up too much space. Don't think any rounding is necessary. The only problem now seems to be that either the axes or the labels are the wrong way around. When moving the mouse vertically, the x-axis value changes and vice versa.

**Pull request merged**